### PR TITLE
Refactor(plugins): Move jinja filter code for arista.avd.convert_dicts to PyAVD

### LINKED ISSUE #4072: Refactor(plugins): Move jinja filter code for arista.avd.convert_dicts to PyAVD
### Enhancement summary

Move jinja filter code for arista.avd.convert_dicts to PyAVD.

### Which component of AVD is impacted

plugins

### Use case example

-

### Describe the solution you would like

`The code and unit tests for convert_dicts filter should be moved to pyavd/j2filters and imported inside plugins/filter using pyavd wrapper.`

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Move jinja filter code for arista.avd.convert_dicts to PyAVD

## Related Issue(s)

Fixes #4072 

## Component(s) name

Filter
- `convert_dict`

## Proposed changes

 Move `convert_dict` code to PyAVD.

## How to test

CI should catch things.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyavd/j2filters/convert_dicts.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    output.append({primary_key: key})
+                else:
+                    # Nested dictionary
+                    output.append(
+                        {
+                            primary_key: key,
+                            **dictionary[key],
+                        }
+                    )
+        return output
```
Comment (gmuloc) [inline python-avd/pyavd/j2filters/convert_dicts.py]: You can reduce the complexity that is present in sonarlint  (not sure it goes below 15 but thats already a thing

```suggestion
    if not isinstance(dictionary, (dict, list)) or os.environ.get("AVD_DISABLE_CONVERT_DICTS"):
        # Not a dictionary/list, return the original
        return dictionary
    if isinstance(dictionary, list):
        output = []
        for element in dictionary:
            if not isinstance(element, dict):
                output.append({primary_key: element})
            elif primary_key not in element and secondary_key is not None:
                # if element of nested dictionary is a dictionary but primary key is missing, insert primary and secondary keys.
                for key in element:
                    output.append(
                        {
                            primary_key: key,
                            secondary_key: element[key],
                        }
                    )
            else:
                output.append(element)
        return output
    # This is now a dict
    output = []
    for key in dictionary:
        if secondary_key is not None:
            # Add secondary key for the values if secondary key is provided
            output.append(
                {
                    primary_key: key,
                    secondary_key: dictionary[key],
                }
            )
        else:
            if not isinstance(dictionary[key], dict):
                # Not a nested dictionary
                output.append({primary_key: key})
            else:
                # Nested dictionary
                output.append(
                    {
                        primary_key: key,
                        **dictionary[key],
                    }
                )
    return output
```

Comment (Vibhu-gslab) [reply python-avd/pyavd/j2filters/convert_dicts.py]: Committed suggestions.

File: python-avd/pyavd/j2filters/convert_dicts.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+import os
+
+
+def convert_dicts(dictionary, primary_key="name", secondary_key=None):
```
Comment (MaheshGSLAB) [inline python-avd/pyavd/j2filters/convert_dicts.py]:
```suggestion
def convert_dicts(dictionary: dict | list, primary_key: str = "name", secondary_key: str | None = None) -> list:
```

File: python-avd/tests/pyavd/j2filters/test_convert_dict.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+# that can be found in the LICENSE file.
+from __future__ import absolute_import, division, print_function
+
+import pytest
+from pyavd.j2filters.convert_dicts import convert_dicts
+
+DEFAULT_PRIMARY_KEY = "name"
+NESTED_LIST_OF_DICT = {
+    "TEST1": [{"type": "permit", "extcommunities": "65000:65000"}, {"type": "deny", "extcommunities": "65002:65002"}],
+    "TEST2": [{"type": "deny", "extcommunities": "65001:65001"}],
```
Comment (MaheshGSLAB) [inline python-avd/tests/pyavd/j2filters/test_convert_dict.py:12]: can we have one more level of nested dict, is this supported?

Comment (MaheshGSLAB) [reply python-avd/tests/pyavd/j2filters/test_convert_dict.py:12]: "TEST1": [{"type": "permit", "extcommunities": "65000:65000", "more": {"data": true}}, {"type": "deny", "extcommunities": "65002:65002"}]

File: python-avd/pyavd/j2filters/convert_dicts.py (hunk 2 above)
Comment (Vibhu-gslab) [reply python-avd/pyavd/j2filters/convert_dicts.py]: Done!

File: python-avd/tests/pyavd/j2filters/test_convert_dict.py (hunk 3 above)
Comment (Vibhu-gslab) [reply python-avd/tests/pyavd/j2filters/test_convert_dict.py:12]: This function will insert primary_key to outer dictionary keys.

**Review by MaheshGSLAB — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]
LGTM

**Pull request merged**